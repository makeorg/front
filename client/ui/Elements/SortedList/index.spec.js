import React from 'react';
import renderer from 'react-test-renderer';
import { SortedList } from './index';

jest.mock('./style', () => ({
  SortedListStyle: 'SortedListStyle',
  SortedButtonStyle: 'SortedButtonStyle',
  SortedRadioStyle: 'SortedRadioStyle',
  CheckedStyle: 'CheckedStyle',
}));

jest.mock('Client/ui/Svg/elements', () => ({
  SvgCheck: 'SvgCheck',
}));

const sortArray = ['foo', 'bar', 'baz'];

describe('SortedList', () => {
  it('must match the snapshot with default Props', () => {
    const component = renderer
      .create(<SortedList currentSort="foo" availableSorts={sortArray} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
