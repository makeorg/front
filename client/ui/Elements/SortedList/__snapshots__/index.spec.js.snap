// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortedList must match the snapshot with default Props 1`] = `
.c0 {
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

<SortedListStyle>
  <li>
    <SortedButtonStyle
      className="selected"
      onClick={[Function]}
    >
      <SortedRadioStyle>
        <SvgCheck
          style="CheckedStyle"
        />
      </SortedRadioStyle>
      <span
        className="c0"
      >
        consultation.sortby
      </span>
      consultation.sort.foo
    </SortedButtonStyle>
  </li>
  <li>
    <SortedButtonStyle
      className={false}
      onClick={[Function]}
    >
      <SortedRadioStyle />
      <span
        className="c0"
      >
        consultation.sortby
      </span>
      consultation.sort.bar
    </SortedButtonStyle>
  </li>
  <li>
    <SortedButtonStyle
      className={false}
      onClick={[Function]}
    >
      <SortedRadioStyle />
      <span
        className="c0"
      >
        consultation.sortby
      </span>
      consultation.sort.baz
    </SortedButtonStyle>
  </li>
</SortedListStyle>
`;
